dotnet/samples/Concepts/AgentSyntax/Getting_Started/Step4_KernelFunctionStrategies.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Threading.Tasks;
+using Examples;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.Agents.Chat;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace GettingStarted;
+
+/// <summary>
+/// Demonstrate usage of <see cref="KernelFunctionTerminationStrategy"/> and <see cref="KernelFunctionSelectionStrategy"/>
+/// to manage <see cref="AgentGroupChat"/> execution.
+/// </summary>
+public class Step4_KernelFunctionStrategies(ITestOutputHelper output) : BaseTest(output)
+{
+    private const string ReviewerName = "ArtDirector";
+    private const string ReviewerInstructions =
+        """
+        You are an art director who has opinions about copywriting born of a love for David Ogilvy.
+        The goal is to determine is the given copy is acceptable to print.
+        If so, state that it is approved.
+        If not, provide insight on how to refine suggested copy without example.
+        """;
+
+    private const string CopyWriterName = "Writer";
+    private const string CopyWriterInstructions =
+        """
+        You are a copywriter with ten years of experience and are known for brevity and a dry humor.
+        You're laser focused on the goal at hand. Don't waste time with chit chat.
+        The goal is to refine and decide on the single best copy as an expert in the field.
+        Consider suggestions when refining an idea.
+        """;
+
+    [Fact]
+    public async Task RunAsync()
+    {
+        // Define the agents
+        ChatCompletionAgent agentReviewer =
+            new()
+            {
+                Instructions = ReviewerInstructions,
+                Name = ReviewerName,
+                Kernel = this.CreateKernelWithChatCompletion(),
+            };
+
+        ChatCompletionAgent agentWriter =
+            new()
+            {
+                Instructions = CopyWriterInstructions,
+                Name = CopyWriterName,
+                Kernel = this.CreateKernelWithChatCompletion(),
+            };
+
+        KernelFunction terminationFunction =
+            KernelFunctionFactory.CreateFromPrompt(
+                $$$"""
+                Determine if the copy has been approved.  If so, respond with a single word: yes
+
+                History:
+                {{${{{KernelFunctionTerminationStrategy.HistoryArgumentName}}}}}
+                """);
```

Comment:
Can we simplify this for the developer?

E.g. how about we allow developers to define the history argument name and then pass their choice into the `KernelFunctionTerminationStrategy` instance. This would make this template simpler and also allow any history argument name to be used.

Also we are using the term `InputVariable` so we should be also using `HistoryVariableName`

Reply:
Good idea, I'll explore this.

Reply:
Updated